### PR TITLE
[Filing] Route protection

### LINKED ISSUE #582: [Routing] Prevent direct navigation via the URL beyond the "next step"
If a user attempts to access a Filing process step (ex. /submit) that is beyond where they've progressed (getFilingSteps(...).nextStepIndex), reroute them to the appropriate place (next incomplete step).

### PR DESCRIPTION
**Pull request by meissadia**
Close #582 

## Changes

- Fix: FilingCreate - Improve method of redirection
- Feat: FilingSteps - Filing route protection
  - If a user tries to access a Filing step beyond where they've progressed in the Filing process, we redirect them to the next step they need to focus on.

## How to test this PR

1. Run through the Filing process for an Institution that has not upload before and verify you can get to the `Submit` stage
1. Run through the Filing process for an Institution that has previously uploaded before and verify you can get to the `Submit` stage
2. Try to go directly reach later stages for an institution that has not reached those stages (i.e. `/submit` for an institution who needs to correct `/errors`) and verify that you are redirected to the appropriate Filing step

## Demo
- User is on the `warnings` step
- User tries to reach the `submit` step but is redirect to `warnings`
- User tires to reach the `contact` step but is redirect to `warnings`
- User is able to directly access the `errors` step since it is complete
- User is able to directly access the `upload` step since it is complete

https://github.com/cfpb/sbl-frontend/assets/2592907/ce4c6065-78ac-4f32-a000-8e3a31597cd1

### CODE REVIEW

File: src/pages/Filing/FilingApp/FilingProtectedRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  if (error) return <Navigate to='/500' state={{ message: error.message }} />;
+
+  const { nextStepIndex } = getFilingSteps(submission, filing);
+
+  const targetPage = location.pathname.split('/').slice(NegativeOne)[Zero];
+  const targetIndex = FILING_PAGE_ORDER.indexOf(targetPage);
+
+  // Redirect
+  if (!targetPage || targetIndex > nextStepIndex) {
```
Comment (billhimmelsbach) [inline src/pages/Filing/FilingApp/FilingProtectedRoute.tsx:40]: Sweet!

File: src/pages/Filing/FilingApp/FilingWarnings/index.tsx [hunk 2]
```diff
@@ -116,12 +118,13 @@ function FilingWarnings(): JSX.Element {
...
-    setFormSubmitLoading(false); // Clear loading indicator
-
     if (isSubmissionAccepted(response)) {
-      setBoxChecked(true);
+      await queryClient.invalidateQueries({
+        queryKey: [`fetch-filing-submission`, lei, year],
+      });
       navigate(nextPage);
     } else {
+      setFormSubmitLoading(false); // Clear loading indicator
```
Comment (billhimmelsbach) [inline src/pages/Filing/FilingApp/FilingWarnings/index.tsx:127]: Also, sweet!

**Review by billhimmelsbach — APPROVED** [review]
Looks good, works great!

**Comment by meissadia** [comment]
Merge conflicts resolved

**Review by billhimmelsbach — APPROVED** [review]

**Pull request merged**